modules/promotion/src/Element/CouponRedemptionForm.php

```diff
@@ -67,11 +72,14 @@ public function getInfo() {
    *   The processed form element.
    */
   public static function processForm(array $element, FormStateInterface $form_state, array &$complete_form) {
-    if (empty($element['#order'])) {
-      throw new \InvalidArgumentException('The commerce_coupon_redemption_form element requires the #order property.');
+    if (empty($element['#order_id'])) {
+      throw new \InvalidArgumentException('The commerce_coupon_redemption_form element requires the #order_id property.');
     }
-    if (!$element['#order'] instanceof OrderInterface) {
-      throw new \InvalidArgumentException('The commerce_coupon_redemption_form #order property must be an order entity.');
+
+    $order_storage = \Drupal::entityTypeManager()->getStorage('commerce_order');
+    $order = $order_storage->load($element['#order_id']);
+    if (!$order instanceof OrderInterface) {
+      throw new \InvalidArgumentException('The commerce_coupon_redemption_form #order_id property must be an existing order entity.');
```

Comment:
This argument message is misleading, need to be clear that we need the ID of an existing order, and not the .order itself. "The commerce_coupon_redemption #order_id must be a valid order ID." ?